### PR TITLE
Fix prediction freeze

### PR DESCRIPTION
**Pull request by eggrobin**

### CODE REVIEW

File: ksp_plugin/vessel_body.hpp [hunk 1]
```diff
@@ -313,12 +314,25 @@ inline void Vessel::FlowProlongation(Instant const& time) {
 
 inline void Vessel::FlowPrediction(Instant const& time) {
   if (time > prediction_->last().time()) {
-    ephemeris_->FlowWithAdaptiveStep(
+    bool finite_time = IsFinite(time - prediction_->last().time());
```
Comment (pleroy) [inline ksp_plugin/vessel_body.hpp]: Isn't this the same as `IsFinite(time)`?  Also, `const`.

File: ksp_plugin/vessel_body.hpp [hunk 2]
```diff
@@ -313,12 +314,25 @@ inline void Vessel::FlowProlongation(Instant const& time) {
...
+    Instant const t = finite_time ? time : ephemeris_->t_max();
+    // This will not prolong the ephemeris if |time| is infinite (but it may do
+    // so if it is finite).
+    bool reached_t = ephemeris_->FlowWithAdaptiveStep(
```
Comment (pleroy) [inline ksp_plugin/vessel_body.hpp]: `const`

File: ksp_plugin_test/vessel_test.cpp [hunk 3]
```diff
@@ -187,5 +192,45 @@ TEST_F(VesselTest, SerializationSuccess) {
...
+  vessel_->set_prediction_adaptive_step_parameters(
+      Ephemeris<Barycentric>::AdaptiveStepParameters(
+          DormandElMikkawyPrince1986RKN434FM<Position<Barycentric>>(),
+          /*max_steps=*/5,
+          /*length_integration_tolerance=*/1 * Metre,
+          /*speed_integration_tolerance=*/1 * Metre / Second));
+  Instant previous_t_max = ephemeris_->t_max();
+  for (int i = 0; i < 10; ++i) {
+    vessel_->UpdatePrediction(t2_ +
+                              std::numeric_limits<double>::infinity() * Second);
```
Comment (pleroy) [inline ksp_plugin_test/vessel_test.cpp]: Do we need `t2_ + `?

File: ksp_plugin_test/vessel_test.cpp [hunk 4]
```diff
@@ -187,5 +192,45 @@ TEST_F(VesselTest, SerializationSuccess) {
...
+TEST_F(VesselTest, PredictBeyondTheInfinite) {
+  vessel_->CreateHistoryAndForkProlongation(t1_, d1_);
+  vessel_->AdvanceTimeNotInBubble(t2_);
+  vessel_->set_prediction_adaptive_step_parameters(
+      Ephemeris<Barycentric>::AdaptiveStepParameters(
+          DormandElMikkawyPrince1986RKN434FM<Position<Barycentric>>(),
+          /*max_steps=*/5,
+          /*length_integration_tolerance=*/1 * Metre,
+          /*speed_integration_tolerance=*/1 * Metre / Second));
+  Instant previous_t_max = ephemeris_->t_max();
```
Comment (pleroy) [inline ksp_plugin_test/vessel_test.cpp]: `const`

File: ksp_plugin_test/vessel_test.cpp [hunk 5]
```diff
@@ -187,5 +192,45 @@ TEST_F(VesselTest, SerializationSuccess) {
...
+  for (int i = 0; i < 10; ++i) {
+    vessel_->UpdatePrediction(t2_ +
+                              std::numeric_limits<double>::infinity() * Second);
+  }
+  // We stop prolonging when the ephemeris gets long enough (in this case, a
+  // single prolongation suffices).
+  EXPECT_THAT(ephemeris_->t_max() - previous_t_max,
+              Le(FlightPlan::max_ephemeris_steps_per_frame *
```
Comment (pleroy) [inline ksp_plugin_test/vessel_test.cpp]: I am le flight plan.

File: ksp_plugin_test/vessel_test.cpp [hunk 6]
```diff
@@ -187,5 +192,45 @@ TEST_F(VesselTest, SerializationSuccess) {
...
+
+  vessel_->set_prediction_adaptive_step_parameters(
+      Ephemeris<Barycentric>::AdaptiveStepParameters(
+          DormandElMikkawyPrince1986RKN434FM<Position<Barycentric>>(),
+          /*max_steps=*/1000,
+          /*length_integration_tolerance=*/1 * Metre,
+          /*speed_integration_tolerance=*/1 * Metre / Second));
+  previous_t_max = ephemeris_->t_max();
+  for (int i = 0; i < 10; ++i) {
+    vessel_->UpdatePrediction(t2_ +
```
Comment (pleroy) [inline ksp_plugin_test/vessel_test.cpp]: Same as above.

File: ksp_plugin/vessel_body.hpp (hunk 1 above)
Comment (eggrobin) [inline ksp_plugin/vessel_body.hpp]: No, and done.

File: ksp_plugin/vessel_body.hpp (hunk 2 above)
Comment (eggrobin) [inline ksp_plugin/vessel_body.hpp]: Done.

File: ksp_plugin_test/vessel_test.cpp (hunk 3 above)
Comment (eggrobin) [inline ksp_plugin_test/vessel_test.cpp]: Yes.

File: ksp_plugin_test/vessel_test.cpp (hunk 6 above)
Comment (eggrobin) [inline ksp_plugin_test/vessel_test.cpp]: Same as above.

**Pull request merged**